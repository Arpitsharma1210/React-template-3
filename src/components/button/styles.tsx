import Button from '@mui/material/Button';
import styled from 'styled-components';

export const StyledButton = styled(Button)<{
  fullWidth?: boolean;
  btnType?: string;
}>`
  padding: ${(props) => (props.btnType === 'primary' ? '8px 18px' : '16px 18px ')} !important;
  white-space: nowrap;
`;
