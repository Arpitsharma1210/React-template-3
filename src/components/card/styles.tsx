import { Card, CardContent } from '@mui/material';
import styled, { css } from 'styled-components';
import { respondTo } from '../../theme/style.layout';
import { brand } from '../../theme/style.palette';

export const StyledCard = styled(Card) <{ bordered?: boolean }>`
    margin-top: 100px;
    ${respondTo.smOnly}{
        margin-top: 0px;
        width:100%;
        height: 100vh;
   }
   ${({ bordered }) => bordered && css`
        border-top : 6px solid ${brand.primaryMain}
   `}
`

export const StyledCardContent = styled(CardContent)`
    ${respondTo.smOnly}{
        padding: 80px 20px !important;
   }
`