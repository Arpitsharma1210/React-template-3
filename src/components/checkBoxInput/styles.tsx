import { Checkbox, FormControlLabel } from "@mui/material";
import { styled } from "styled-components";
import { baseFontFamily, fontSize, fontWeight } from "../../theme/style.typography";
import { brand, colors } from "../../theme/style.palette";



export const StyledFormControlLabel = styled(FormControlLabel)`
    margin-left : 0px !important;
    .MuiFormControlLabel-label {
        font-size: ${fontSize.b2};
        font-weight: ${fontWeight.regular};
        font-family: ${baseFontFamily};
        color: ${brand.textColourDark};
    }
`

export const StyledCheckbox = styled(Checkbox)`

`