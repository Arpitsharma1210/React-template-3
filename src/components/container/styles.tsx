import { Container } from '@mui/material';
import styled from 'styled-components';
import { respondTo } from '../../theme/style.layout';

export interface StyledContainerProps {
    centerAlign?: boolean;
}

export const StyledContainer = styled(Container) <StyledContainerProps>`
    align-items : ${({ centerAlign }) => centerAlign ? 'center' : 'baseline'} !important;
    ${respondTo.smOnly}{
        padding: 0 !important;
    }
`