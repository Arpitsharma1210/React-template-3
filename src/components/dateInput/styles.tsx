import { DatePicker } from "@mui/x-date-pickers";
import { styled } from "styled-components";
import { fontSize, fontWeight } from "../../theme/style.typography";
import { brand } from "../../theme/style.palette";


export const StyledDatePicker = styled(DatePicker)<{fullWidth?:boolean; }>`

    .MuiInputBase-input{
        padding: 10px 16px;
        font-size: ${fontSize.b2};
        font-weight: ${fontWeight.regular};
        color: ${brand.inputBorder};
        &::placeholder {
            color: ${brand.placeHolder};
        }
    }
   
`