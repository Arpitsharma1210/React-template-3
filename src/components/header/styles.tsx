import { styled } from 'styled-components';
import { Typography } from '@mui/material';
import { colors } from '../../theme/style.palette';
import { respondTo } from '../../theme/style.layout';

export const StyledContainer = styled.div`
    display: flex;
    align-items:center;
    flex: 1;
    margin: 16px 0px;
    width: 100%;
`

export const StyledLogoContainer = styled.div`
    margin-right: 32px;
    display: flex;
    ${respondTo.smOnly}{
        margin-left: 32px;
    }
`
export const StyledLogo = styled.img`
    width : 34px;

`

export const StyledNavContainer = styled.div`
    display: flex;
    align-items:center;
    flex: 1;
`

export const StyledNavItem = styled.div<{active?:boolean}>`
    margin-right: 24px;
    padding : 8px;
    border-radius: 8px;
    cursor : pointer;
    background-color: ${({active})=> active ? colors.greyLight : 'transparent'}
`

export const StyledNavItemText = styled(Typography)`

`
export const StyledLogoutContainer= styled.div`
    margin-right: 16px;
    cursor : pointer;
`

export const StyledLogoutText = styled(Typography)`

`