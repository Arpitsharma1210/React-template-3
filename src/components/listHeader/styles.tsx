import { css, styled } from 'styled-components';
import { Typography } from '@mui/material';
import { brand, colors } from '../../theme/style.palette';
import { respondTo } from '../../theme/style.layout';
import { fontSize } from '../../theme/style.typography';

export const StyledContainer = styled.div`
    display: flex;
    align-items: center;
    width: 100%;
    margin: 16px 0px;
    ${respondTo.smOnly}{
        flex-direction: column;
        align-items: baseline;
    }
`

export const StyledHeadingContainer = styled.div`
    margin-right: 32px;
    display: flex;
    ${respondTo.smOnly}{
        margin-left: 32px;
        margin-bottom: 8px;
    }
`
export const StyledHeading = styled(Typography)`
  
`

export const StyledActionItemContainer = styled.div`
    display: flex;
    flex: 1;
    align-items: center;
    justify-content: flex-end;
    ${respondTo.smOnly}{
        flex-direction: column;
        align-items: baseline;
    }
`

export const StyledActionItem = styled.div<{lastItem?:boolean}>`
    margin-right: 24px;
    ${({lastItem})=>lastItem&&css`
        margin-right: 0px;
    `}
    ${respondTo.smOnly}{
        margin: 8px 0px;
    }
`

export const SearchCtaContainer = styled.div`
    display: flex;
    align-items: center;
    padding : 8px 16px;
    cursor:pointer;
    ${respondTo.smOnly}{
        padding-left: 0px;
        margin-bottom: 2px;
    }
`
export const StyledCtaText = styled(Typography)`
  
`

export const SearchInputContainer = styled.div`
    display: flex;
    align-items: center;
    border: 1px solid ${colors.greyMedium};
    border-radius: 8px;

`

export const StyledSearchInput = styled.input`
    min-width: 240px;
    padding : 8px 0px;
    border: none;
    outline: none;
    font-size:${fontSize.b2};
    line-height: 24px;
    &::placeholder {
        color:  ${colors.grey};
    }
`

export const StyledSearchClear = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    background-color: ${colors.greyLight};
    border-top-right-radius: 8px;
    border-bottom-right-radius: 8px;
    padding: 10px 8px;
`