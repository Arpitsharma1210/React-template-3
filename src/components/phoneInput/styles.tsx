import PhoneInput from 'react-phone-input-2'
import 'react-phone-input-2/lib/bootstrap.css'
import { css, styled } from 'styled-components';
import { baseFontFamily, fontSize, fontWeight } from '../../theme/style.typography';
import { brand, colors } from '../../theme/style.palette';

export const StyledPhoneInput = styled(PhoneInput)<{error?: boolean; fullWidth?:boolean; readOnly?:boolean;}>`

    .form-control {
        width:100%;
        max-width : ${({fullWidth})=>fullWidth ? '100%' : '384px'};
        outline:none;
        border: 1px solid ${brand.inputBg};
        background-color: ${brand.inputBg};
        border-radius:6px;
        padding: 10px 16px 10px 56px;
        font-size: ${fontSize.b2};
        font-weight: ${fontWeight.regular};
        font-family: ${baseFontFamily};
        color: ${brand.inputBorder};
        &:hover {
            border-color: ${brand.inputBorder};
        }
        &:focus {
            border-color: ${brand.inputBorder};
            background-color: ${colors.white};
            color: ${brand.textColourDark};
            box-shadow: none;
        }

        ${({ error }) => error && css`
            background-color: ${brand.errorBg};
            &:hover {
                border-color: ${brand.errorBorder};
            }
            &:focus {
                border-color: ${brand.errorBorder};
            }
        `}
        ${({ readOnly }) => readOnly && css`
            background-color: transparent;
            cursor: pointer;
            &:hover {
                border-color: ${brand.inputBg};
            }
            &:focus {
                border-color: ${brand.inputBg};
            }
        `}
    }
    .selected-flag.open{
        &:before {
            box-shadow: none;
            border : none;
            outline : none;
        }
    }
    .selected-flag {
        &:focus:before{
            box-shadow: none;
            border : none;
            outline : none;
        }
        outline : none !important;
    }
 
   
`