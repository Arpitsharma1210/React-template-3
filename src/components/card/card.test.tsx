import React from "react";
import { render, } from "@testing-library/react";
import Card from "./index";
import { StyledCard, StyledCardContent } from "./styles";

describe("Card", () => {
  it("renders without crashing", () => {
    render(<Card children={[]}/>);
    expect(<Card children={[]}/>).toMatchSnapshot();
  });
});

  describe('Styled Components', () => {
    it('renders StyledHeader correctly', () => {
      const wrapper = render(<StyledCard />);
      expect(wrapper).toMatchSnapshot();
    });
    it('renders StyledHeader correctly', () => {
      const wrapper = render(<StyledCardContent  />);
      expect(wrapper).toMatchSnapshot();
    });
  });