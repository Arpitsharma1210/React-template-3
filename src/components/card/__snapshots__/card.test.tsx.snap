// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders without crashing 1`] = `<Card />`;

exports[`Styled Components renders StyledHeader correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bcHOTY hbsRQX css-bhp9pd-MuiPaper-root-MuiCard-root"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root sc-bcHOTY hbsRQX css-bhp9pd-MuiPaper-root-MuiCard-root"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Styled Components renders StyledHeader correctly 2`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiCardContent-root sc-gsMHZj elSXYf css-46bh2p-MuiCardContent-root"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiCardContent-root sc-gsMHZj elSXYf css-46bh2p-MuiCardContent-root"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
